src/contracts/GPv2Settlement.sol

```diff
@@ -353,6 +355,7 @@ contract GPv2Settlement is GPv2Signing, ReentrancyGuard {
             // solhint-disable-next-line not-rely-on-time
             require(validTo < block.timestamp, "GPv2: order still valid");
             filledAmount[orderUid] = 0;
+            preSignature[orderUid] = false;
```

Comment:
I'd expect that pre-signing will be uncommon enough that the gain from freeing pre-signature storage will be less than what we gain from not wasting gas when clearing order refunds.
2000 gas should be about one fifth of the amount of gas we get back from gas refunds, right? So if we expect 1 user out of 5 or less to use pre-signatures, this should go.

```suggestion
            filledAmount[orderUid] = 0;
```

Reply:
I just wanted to be a good Ethereum citizen :sob:.

Joking aside, if we implement "smart orders" for things like IDOs, CMMs, then I think this will become more prominent, and would have the cool effect of having the pre-signature set as a pre-interaction and then have the gas refund happen at the end of the settlement.

Reply:
> having the pre-signature set as a pre-interaction and then have the gas refund happen at the end of the settlement.

The gas refund in this case could not happen with the standard gas refund in a settlement, because in order to be executed an order must be valid and in order to be refunded an order must be invalid. So it would be in a post-interaction anyway.

```
struct OrderRefund {
    bool freePreSignature
    bytes orderUid
}
```

In principle it makes sense. I'm worried about gas consumption. how much do we get back from refunds after the refactoring? I wonder how much space for gas inefficiencies we have left in the 15k gas refund.

Reply:
How about making the order refund be something like

```
struct OrderRefunds {
  bytes eoaOrderUids,
  bytes presignOrderUids
}
```
 and only clear preSignature for the latter.

Reply:
Nice! Thats a constant 32 bytes of 0s per settlement instead of one per refund.

Reply:
Implemented in #409